src/mistral-client.ts

```diff
@@ -1,132 +1,121 @@
     const fetchFunc = await configuredFetch;
-    return fetchFunc(...args);
+    return fetchFunc(input, init);
   }
 
   /**
-   *
-   * @param {*} method
-   * @param {*} path
-   * @param {*} request
-   * @param {*} signal
-   * @return {Promise<*>}
+   * Creates a chat completion request
+   * @param param0 
+   * @returns {MistralChatCompletionRequest}
+   */
+  private _makeChatCompletionRequest({
+    model,
+    messages,
+    tools,
+    temperature,
+    maxTokens,
+    topP,
+    randomSeed,
+    stream,
+    safeMode,
+    safePrompt,
+    toolChoice,
+    responseFormat,
+  }: any) {
+    // if modelDefault and model are undefined, throw an error
+    if (!model && !this.modelDefault) {
+      throw new MistralAPIError('You must provide a model name');
+    }
+    return {
+      model: model ?? this.modelDefault,
+      messages: messages,
+      tools: tools ?? undefined,
+      temperature: temperature ?? undefined,
+      max_tokens: maxTokens ?? undefined,
+      top_p: topP ?? undefined,
+      random_seed: randomSeed ?? undefined,
+      stream: stream ?? undefined,
+      safe_prompt: (safeMode || safePrompt) ?? undefined,
+      tool_choice: toolChoice ?? undefined,
+      response_format: responseFormat ?? undefined,
+    } as MistralChatCompletionRequest;
+  }
+
+  /**
+   * 
+   * @param method 
+   * @param path 
+   * @param request 
+   * @param signal 
+   * @returns 
    */
-  _request = async function(method, path, request, signal) {
+  private async _request(method: string, path: string, request?: any, signal?:AbortSignal): Promise<any> {
     const url = `${this.endpoint}/${path}`;
-    const options = {
-      method: method,
+    const options: RequestInit = {
+      method,
       headers: {
         'User-Agent': `mistral-client-js/${VERSION}`,
-        'Accept': request?.stream ? 'text/event-stream' : 'application/json',
+        Accept: request?.stream ? 'text/event-stream' : 'application/json',
         'Content-Type': 'application/json',
-        'Authorization': `Bearer ${this.apiKey}`,
+        Authorization: `Bearer ${this.apiKey}`,
       },
       body: method !== 'get' ? JSON.stringify(request) : null,
-      signal: combineSignals(
-        [AbortSignal.timeout(this.timeout * 1000), signal]),
+      signal: signal? combineSignals([AbortSignal.timeout(this.timeout * 1000), signal]): AbortSignal.timeout(this.timeout * 1000),
     };
 
     for (let attempts = 0; attempts < this.maxRetries; attempts++) {
       try {
         const response = await this._fetch(url, options);
-
         if (response.ok) {
           if (request?.stream) {
             // When using node-fetch or test mocks, getReader is not defined
-            if (typeof response.body.getReader === 'undefined') {
+            if (typeof response.body!.getReader === 'undefined') {
               return response.body;
-            } else {
-              const reader = response.body.getReader();
+            }else {
```

Comment:
Looks like formatting needs running? `}els {`

Reply:
this one I think it might need something like `.prettierrc` config to align formatting afterwards. 
I have noticed that there was a pr related with single-quote.